app/gui2/src/components/lexical/sync.ts

```diff
@@ -0,0 +1,46 @@
+import { $setRootText } from '@/components/lexical/utils'
+import { $getRoot, type EditorState } from 'lexical'
+import { useLexicalComposer } from 'lexical-vue'
+import { computed, ref } from 'vue'
+
+const SYNC_TAG = 'ENSO_SYNC'
+
+function $getRootText() {
+  return $getRoot().getTextContent()
+}
+
+export function useSync(
```

Comment:
This function's is very generic. Maybe rename it to `useLexicalSync` for a bit more context, and add some docs.